CONTRIBUTING.md

```diff
@@ -109,6 +106,66 @@ The contribution process is outlined below:
    maintainer. At this point, a Meta employee will be notified to merge your PR,
    due to tooling limitations.
 
+## Commit Messages
+
+We build Velox for the long-run, and to do so it is crucial that project
+maintainers are able to efficiently inspect and understand project logs.
+
+Commit messages that follow a strict pattern improve maintainability, and allow
+tasks such as summarizing changelogs and identifying API breaking changes to be
+automated. Despite requiring more rigor from authors and reviewers, consistency
+creates productivity in the long term.
+
+In Velox, commit messages are generated based on the input provided to PRs, and
+must follow the [conventional commit
+specification](https://www.conventionalcommits.org/en/v1.0.0/) in the following
+manner:
+
+**PR titles** must follow the pattern:
+
+> \<type\>[optional scope]: \<description\>
+
+where:
+
+* *Type* can be any of the following keywords:
+  * **feat** when new features are being added.
+  * **fix** for bug fixes.
+  * **build** for build or CI-related improvements.
+  * **docs** for enhancements to documentation (only).
```

Comment:
Can we change this to 'doc' for consistency?

Reply:
Curious, what should we use for PRs that do not fit any of these categories (say, adding tests) or PRs that have multiple of these?

Reply:
For those `chore` is usually used but in the discussion it was pointed out that the word has negative connotations and was renamed to refactor. While I see that point I would agree that we need another bucket, maybe 'misc' or explicitly list them under refactor

Reply:
We could also create a "tests" category. 

> ... or PRs that have multiple of these?

I think in that case you would chose the "main" one. Like if you're adding a new feature with tests and docs, probably call out as a feature. But in general, if you have more than one "main" category, maybe you should have submitted them as different PRs in the first place?

Reply:
Can we make `docs` -> `doc`?